File: cryptography-cipher/src/test/java/dev/retrotv/crypto/cipher/generator/IVGeneratorTest.java
package dev.retrotv.crypto.cipher.generator;

import dev.retrotv.crypto.cipher.enums.ECipher;
import dev.retrotv.crypto.cipher.enums.EMode;
import dev.retrotv.crypto.exception.GenerateException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class IVGeneratorTest {
    @Test
    @DisplayName("Exception 발생 테스트")
    void test_generateException() {
        assertThrows(IllegalArgumentException.class, () -> IVGenerator.generateIV(4));
        assertThrows(GenerateException.class, () -> IVGenerator.generateIV(ECipher.AES, EMode.ECB));
        assertThrows(GenerateException.class, () -> IVGenerator.generateIV(ECipher.DES, EMode.CCM));
    }

    @Test
    @DisplayName("iv 생성 테스트")
    void test_generateKey() {
        byte[] key = IVGenerator.generateIV(ECipher.AES, EMode.CBC);
        assertEquals(16, key.length);

        key = IVGenerator.generateIV(ECipher.AES, EMode.CCM);
        assertEquals(12, key.length);

        key = IVGenerator.generateIV(ECipher.DES, EMode.CBC);
        assertEquals(8, key.length);
    }
}

File: cryptography-chiper/src/test/kotlin/dev/retrotv/crypto/encryption/generator/IVGeneratorTest.kt
package dev.retrotv.crypto.encryption.generator

import dev.retrotv.crypto.enums.ECipher.AES
import dev.retrotv.crypto.enums.ECipher.DES
import dev.retrotv.crypto.enums.EMode.*
import dev.retrotv.crypto.exception.GenerateException
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.DisplayName
import kotlin.test.Test

class IVGeneratorTest {

    @Test
    @DisplayName("Exception 발생 테스트")
    fun test_generateException() {
        assertThrows(IllegalArgumentException::class.java) {
            generateIV(4)
        }

        assertThrows(GenerateException::class.java) {
            generateIV(AES, ECB)
        }

        assertThrows(GenerateException::class.java) {
            generateIV(DES, CCM)
        }
    }

    @Test
    @DisplayName("iv 생성 테스트")
    fun test_generateKey() {
        var key = generateIV(AES, CBC)
        assert(key.size == 16)

        key = generateIV(AES, CCM)
        assert(key.size == 12)

        key = generateIV(DES, CBC)
        assert(key.size == 8)
    }
}